### DISCUSSION TITLE
Week 11 check in

### DISCUSSION THREAD

**Discussion by minhphan03** [UW-Upwelling-Project/Summer-2023#16] [General]
________________

**Comment by minhphan03** [comment]
Hi @eeholmes I had such a hard time with the AWS customer support that I decided maybe not to work with it anymore and thinking maybe we can move to azure instead? I already had an account with them for my database class last year, and also had the files transferred to azure blob (equivalent to s3) and thinking maybe we can start doing something with it? 

Pros: more user-friendly, smoother transitioning
Cons: less technical support/documentation from users, cannot upload zarr directly

**Reply by minhphan03** [reply]
# Minh's Monday Check-in

I mulled over today and we have two options:
1. Continue to work on AWS using your account (you can create a secondary IAM role giving me specific permissions on s3/lambda/sts) and let me work with it until the internship is over. I cannot create another account because I only have one phone number to verify (and they would ask for like phone bills, credit card statements, etc. to verify if they suspect anything)
2. Use Azure: I worked out today how to upload zarr files to azure directly with python. we can use the azure python library to generate sas tokens for users in a certain region (maybe us-west-2 is oregon & washington) as they send requests with public ip address. Maybe we can run this address through an ip region finder (more research on this) and add an if statement to [return credentials](https://learn.microsoft.com/en-us/azure/storage/blobs/sas-service-create-python). I tried deploying a azure function app today and it worked ok so maybe we can do that...? Check out this sample function app that return a name that I deployed https://minhphan0612.azurewebsites.net/api/HttpTrigger1?name=phan (change name= whatever you like)

Of course the first option is more ideal to our initial plan, but let's only do that if you're comfortable. I think the second one is cool, too. After all it all boils down to how we can define the region (us-west-2) a user is in. My Azure is under UW license so I can use it for free as a student, so cost is not really an issue.

**Comment by NaNa7Miiii** [comment]
Jiarui Monday Checkin:
1. Made predictions using linear regression (for one specific location such as lat=45, lon=-45)
Using MUR Data and SST only:
```
Predictions: [294.52145679 294.58229795 294.6431391  294.70398025 294.7648214 ]
Actual Values: [292.749   292.72998 292.69    291.211   291.021  ]
Mean Absolute Error: 2.562945006778148
```
Using Our Zarr Data and ['CHL', 'air_temp', 'sst', 'wind_dir', 'wind_speed']:
```
        Date  Predictions   True Data
0 2021-10-01   299.626617  299.786469
1 2021-10-02   297.302155  300.218750
2 2021-10-03   297.312805  300.541656
3 2021-10-04   297.312775  300.343750
4 2021-10-05   297.312775  300.671875
Mean Squared Error: 7.885585784912109
```
The results from linear regression are not good, indicating it's necessary to use more complex models such as our ConvLSTM model.

2. Made a customized model arch that can accept any input shape, not just limited to (401, 501). 
<img width="923" alt="Screenshot 2023-08-28 at 3 48 23 PM" src="https://github.com/UW-Upwelling-Project/Summer-2023/assets/97627889/89d3de3b-f3bc-42f7-9661-ebc4143719ec">

**Comment by minhphan03** [comment]
# Tuesday's Check-in:
- Developed azure function to handle a GET api call to return temporary storage access signature. I had a hard time struggling with it because apparently read and list are supposed to be two different types of permission. I wrote a notebook in the 2i2c hub (azure-notebooks/blob-access.ipynb) to give some details on how to generate tokens!

**Comment by NaNa7Miiii** [comment]
Tuesday's Check in:
Deploy my streamlit app on the cloud

**Comment by NaNa7Miiii** [comment]
Jiarui Wed & Thursday checkin:
* finished deploying my current streamlit app:
github link: https://github.com/NaNa7Miiii/upwelling_prediction
streamlit cloud link: https://upwellingprediction-gkk5gw7pcvdgt73sdmhx9k.streamlit.app/

& Tried to train the upwelling classification model using the mur dataset, but failed due to limited ram.
